### PR TITLE
name constraints - support IP addresses with netmask

### PR DESCRIPTION
**Pull request by reaperhulk**
refs #1947

### CODE REVIEW

File: src/cryptography/hazmat/backends/openssl/x509.py [hunk 1]
```diff
@@ -125,13 +125,30 @@ def _decode_general_name(backend, gn):
...
+            # ipaddress doesn't support packed bytes + netmask. Additionally,
+            # IPv6Network can only handle CIDR rather than the full 16 byte
+            # netmask. To handle this we convert the netmask to integer, then
+            # find the first 0 bit, which will be the prefix. If another 1
+            # bit is present after that the netmask is invalid.
+            base = ipaddress.ip_address(data[:data_len // 2])
+            netmask = utils.int_from_bytes(data[data_len // 2:], 'big')
+            bits = bin(netmask)[2:]
+            prefix = bits.find('0')
+            if bits[prefix:].find('1') != -1:
```
Comment (alex) [inline src/cryptography/hazmat/backends/openssl/x509.py]: Is this just `if "1" not in bits[prefix:]`? Also, `bits.find("0")` returns `-1` on non-existennce. Will that trigger a bug?

Comment (reaperhulk) [inline src/cryptography/hazmat/backends/openssl/x509.py]: Yeah it is. And definitely the former is a bug. I'll have to add one more vector for that.

**Comment by alex** [comment]
tests are not passing

**Comment by alex** [comment]
this needs a rebase

File: src/cryptography/hazmat/backends/openssl/x509.py [hunk 2]
```diff
@@ -141,11 +141,34 @@ def _decode_general_name(backend, gn):
...
         return x509.RegisteredID(x509.ObjectIdentifier(oid))
     elif gn.type == backend._lib.GEN_IPADD:
-        return x509.IPAddress(
-            ipaddress.ip_address(
-                _asn1_string_to_bytes(backend, gn.d.iPAddress)
-            )
-        )
+        data = backend._ffi.buffer(
+            gn.d.iPAddress.data, gn.d.iPAddress.length
+        )[:]
```
Comment (alex) [inline src/cryptography/hazmat/backends/openssl/x509.py]: This should be `_asn1_string_to_bytes`

**Pull request merged**